packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -62,6 +64,16 @@ export abstract class DeployRetrieveExecutor<
     try {
       const components = await this.getComponents(response);
 
+      // concrete classes may have purposefully changed the api version.
```

Comment:
I find this documentation a bit confusing. When you say concrete classes - what exactly are you referring to? Is this in reference to the manifest file?

Reply:
By concrete classes I'm referring to implementations of the abstract class. So if getComponents changes the api version on the component set in the child class, this code respects that. Particularly this is for the manifest-based classes, but if anything extends this base class and manually sets the api version this will work.